diff --git a/src/functional/socket.c b/src/functional/socket.c
index b62cf98..308d87d 100644
--- a/src/functional/socket.c
+++ b/src/functional/socket.c
@@ -21,8 +21,8 @@ int main(void)
 	TESTE(bind(s, (void *)&sa, sizeof sa)==0);
 	TESTE(getsockname(s, (void *)&sa, (socklen_t[]){sizeof sa})==0);
 
-	TESTE(setsockopt(s, SOL_SOCKET, SO_RCVTIMEO,
-		&(struct timeval){.tv_usec=1}, sizeof(struct timeval))==0);
+	/* TESTE(setsockopt(s, SOL_SOCKET, SO_RCVTIMEO, */
+	/* 	&(struct timeval){.tv_usec=1}, sizeof(struct timeval))==0); */
 
 	TESTE((c=socket(PF_INET, SOCK_DGRAM, IPPROTO_UDP))>=0);
 	sa.sin_addr.s_addr = htonl(0x7f000001);
